### ISSUE TITLE
[Enhancement] Implement authentication in DataCubeServer

### ISSUE THREAD

**Issue by mmebsout** [MizarWeb/MizarWeb.github.io#37]
Put back DataCubeServer authentication feature the right way. 
Authentication has been disabled in DataCubeServer with [this commit](https://github.com/MizarWeb/DataCubeServer/commit/ba0fb31d097e8ac13f1c392306d76acda307acd5).

The authentication did not make any difference between public or private folders depending on the authenticated profile.

**Comment by J-Christophe** [comment]
The authorization seems to be performed at the client side (https://github.com/MizarWeb/DataCube/blob/291d206c187f5c25f9fa21ca9530ddffa4ed28cb/src/app/home/dataCube/dataCube.component.ts#L132-L143) instead of the server side (I do not see any control of authorization). This is not nice.

**Comment by mmebsout** [comment]
@J-Christophe Sorry I didn't mean to assign you. I'm sure you have a lot to do. We are going to use it on public files so far so it's not urgent on our side but should definitely not be forgotten hence the ticket :)

**Comment by mmebsout** [comment]
@J-Christophe There is some check on the Server if the property no_credentials_needed is false 
https://github.com/MizarWeb/DataCubeServer/commit/ba0fb31d097e8ac13f1c392306d76acda307acd5 but there is no check whereas the profile has access to this or that folder.

**Comment by J-Christophe** [comment]
Agree, only on authenticate resource there is a check. But the others
resources, nothing. Actually, the authenticate resource should return a
json web token (https://jwt.io/).Then this token should be sent by the gui.
The code for each resource (except authenticate) retrieves the token from
the http header and  uncrypt the token to find the username. Based on the
username, a logic must be implemented to allow or not this user

**Comment by mmebsout** [comment]
Ah yes I remember. yep this is wrong.
